DataFormats/PatCandidates/src/CovarianceParameterization.cc

```diff
@@ -0,0 +1,225 @@
+#include "DataFormats/PatCandidates/interface/CovarianceParameterization.h"
+#include "DataFormats/PatCandidates/interface/liblogintpack.h"
+#include "DataFormats/PatCandidates/interface/libminifloat.h"
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include <boost/format.hpp>
+#include <iostream>
+#include <TParameter.h>
+#include <TVector.h>
+#include <TFolder.h>
+
+uint16_t CompressionElement::pack(float value, float ref) const
+{
+    float toCompress=0;
+    switch(target) {
+        case(realValue):
+          toCompress=value;
+          break;
+        case(ratioToRef):
+          toCompress=value/ref;
+          break;
+        case(differenceToRef):
+          toCompress=value-ref;
+          break;
+    }
+    switch(method) {
+        case(float16):
+          return MiniFloatConverter::float32to16(toCompress*params[0]);
+          break;
+        case(reduceMantissa):
+          return MiniFloatConverter::reduceMantissaToNbits(toCompress,params[0]);
+          break;
+        case(zero):
+          return 0;
+          break;
+        case(one):
+          return 1.0;
+          break;
+          //return  pack16log(toCompress,params[0],params[1],params[2]);
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

.. same about L44